scRNA-seq/07-gene_set_enrichment_analysis.Rmd

```diff
@@ -0,0 +1,352 @@
+    toc_float: true
+author: CCDL for ALSF
+date: 2021
+---
+
+## Objectives
+
+This notebook will demonstrate how to:
+
+- Prepare tabular data of gene-level statistics for use with Gene Set Enrichment Analysis (GSEA)
+- Access [Molecular Signatures Database gene set collections](https://www.gsea-msigdb.org/gsea/msigdb/collections.jsp) via the `msigdbr` package
+- Perform GSEA with the `clusterProfiler` package
+- Visualize GSEA results with the `enrichplot` package
+
+---
+
+In this notebook, we'll analyze the marker genes from cluster 1, just as we did in the previous notebook.
+Unlike ORA, GSEA allows us to use the full list of genes we could reliably measure, rather than picking some cutoff ourselves.
+
+GSEA is a functional class scoring (FCS) approach to pathway analysis that was first introduced in [Subramanian _et al._ (2005)](https://doi.org/10.1073/pnas.0506580102).
+The rationale behind FCS approaches is that small changes in individual genes that participate in the same biological process or pathway can be significant and of biological interest.
+FCS methods are better suited for identifying these pathways that show coordinated changes than ORA.
+In ORA, we pick a cutoff that _typically_ only captures genes with large individual changes.
+
+There are 3 general steps in FCS methods ([Khatri _et al._ 2012]( https://doi.org/10.1371/journal.pcbi.1002375)):
+ 
+1. Calculate a gene-level statistic (here, we'll use the summary log fold changes in our marker genes table)
+2. Gene-level statistics are aggregated into a pathway-level statistic
+3. Assess the statistical significance of the pathway-level statistic
+
+We'll note here that GSEA was designed to detect small coordinated changes between _conditions_ in (necessarily bulk!) microarray data.
+It may be more difficult to suss out small coordinated changes when we assay individual cells, due to technical or biological dropout. 
+
+Individual cells may also differently sized (Blasi *et al.* 2017), be at different stages of the cell cycle ([Satija lab](https://satijalab.org/seurat/archive/v3.1/cell_cycle_vignette.html)), or experiencing different levels of cellular stress ([Luecken and Theis. 2019](https://doi.org/10.15252/msb.20188746)), some of which may be interesting in the context of cancer and therefore should not be corrected for.
+(In practice, it is common to correct for cell cycle stage but we have not done that upstream.)
+These aspects also have the potential to cloud our ability to detect small coordinated changes in other pathways.
+
+There is some literature that suggests that dropout (excessive zeros) observed in single cell data is a function of cell type heterogeneity ([Kim *et al.* 2020](https://doi.org/10.1186/s13059-020-02096-y)). 
+We're using GSEA to compare clusters of cells, which we (ideally) expect to be different biological states and/or cell types. 
+If many genes "drop out" between cell types, we may not expect to capture subtle signals between clusters.
```

Comment:
A thought I have here is that the comparison we are looking at is in some ways a pseudo-bulk comparison in that we are looking at logfold changes between groups of cells. 

I don't know if there is any shrinkage applied in the LFC estimates from `findMarkers()` (I suspect not?), so while it is certainly noisy, my sense is we might overstate some effects as well as miss others.  Which is to say that I agree with the caveats, but might be a bit more ambiguous about directions of effect.

Reply:
How about something like:

> If many genes "drop out" between cell types, we may expect our statistics used as input to GSEA to be noisy.

Reply:
👍🏼